tests/test_triangle.cpp

```diff
@@ -25,6 +26,28 @@ TEST_CASE("Test Triangle normal", "[triangle]")
     }
 }
 
+TEST_CASE("Test subdivision of triangle", "[triangle]")
+{
+    static constexpr int NUM_SAMPLES = 100;
+
+    for (int j = 0; j < NUM_SAMPLES; ++j) {
+        Triangle triangle = random_triangle();
+
+        auto triangles = subdivide4(triangle);
+
+        // Verify that all triangles have the same size.
+        REQUIRE(triangles[0].area() == Approx(triangles[1].area()));
+        REQUIRE(triangles[1].area() == Approx(triangles[2].area()));
+        REQUIRE(triangles[2].area() == Approx(triangles[3].area()));
+        REQUIRE(triangles[3].area() == Approx(triangles[0].area()));
+
+        // Verify sum of sub triangles to be full area.
+        float summed_area = triangles[0].area() + triangles[1].area() +
+            triangles[2].area() + triangles[3].area();
+        REQUIRE(summed_area == Approx(triangle.area()));
```

Comment:
Das reicht leider nicht. Wir müssen tatsächlich wissen, dass jede Fläche 1/4 ist. Diese Annahme ist ueberall in Radiosity.